pkg/util/clusteragent/clusteragent.go

```diff
@@ -0,0 +1,223 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package clusteragent
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+)
+
+/*
+Utils to query the Datadog Cluster Agent (DCA) API
+*/
+
+const (
+	authorizationHeaderKey        = "Authorization"
+	clusterAgentAuthTokenMinLen   = 32
+	clusterAgentAuthTokenFilename = "dca_auth_token"
+)
+
+var globalClusterAgentUtil *DCAUtil
+
+type serviceNames []string
+
+// DCAUtil is required to query the API of Datadog cluster agent
+type DCAUtil struct {
```

Comment:
any reason to call this `Util` instead of `Client`?

Reply:
Indeed, I picked `Client` in the first place.
I fallback to `Util` to be consistent with the others utils...